pkg/calculator/handler.go

```diff
@@ -16,15 +18,63 @@ func NewEmissionCalculator(eventBus bus.Bus) *EmissionCalculator {
 	}
 }
 
-func (c *EmissionCalculator) Apply(event bus.Event) {
+func (ec *EmissionCalculator) Apply(event bus.Event) {
 	// Make sure we got the right event
 	if metricsCollected, ok := event.(v1.MetricsCollected); ok {
-		// TODO: remove this, which is only for debugging purposes
 		instance := metricsCollected.Instance
+		cfg := config.AppConfig().ProvidersConfig
 
-		// TODO: do the calculation
+		emFactors, err := factors.GetEmissionFactors(instance.Provider(), cfg.FactorsDataPath)
+		if err != nil {
+			klog.Errorf("error getting emission factors: %+v", err)
+			return
+		}
+
+		specs, ok := emFactors.Embodied[instance.Kind()]
+		if !ok {
+			klog.Errorf("error finding instance: %s in factor data", instance.Name())
+			return
+		}
+
+		mCPU, ok := instance.Metrics()["cpu"]
```

Comment:
I suggest to use: `mCPU, ok := instance.Metrics()[v1.CPU.String()]`

Just in case in the future we might think about changing the string representation of the CPU resource